Adding "how to create new page" to `CONTRIBUTING.md` (#1294)
ipfs/ipfs-docs: CONTRIBUTING.md

### Discuss

You can also contribute to the project without specifically writing content for the site. Many discussions are happening all around the project. Protocol Labs, and by extension, IPFS, is a collaborative company that relies heavily on the community. We need your opinions and input! If you have an idea about a topic or want to weigh in on a discussion, feel free to have your say on either [GitHub](https://github.com/ipfs/ipfs-docs/issues) or on the [IPFS forums](https://discuss.ipfs.tech/).

> ## Create a new page
> 
> Sometimes, you will need to make a whole new page for content to live. The process to do this is outlined below.
> 
> These steps were performed in Visual Studio Code (VSCode).
> 
> It's important to know what kind of content you are creating. The questions below will help you decide where your new content should live. For example, you wouldn't want to put a page written for experienced users between the already existing **Install** and **Basics** sections.
> 
> - What type of content am I creating?
> - Who is my intended audience?
> - Does this content require a new page or can it be added to an existing page?
> - If a new page or section must be added, how will it fit into the existing structure and flow of the docs?
> 
> When you've answered the questions outlined above and you know where the new content will live, continue with the steps below.

Reviewer: ```suggestion
## Create a new page

Sometimes, you will need to make a whole new page for content to live. The process to do this is outlined below.

These steps were performed in Visual Studio Code (VSCode).

It's important to know what kind of content you are creating. The questions below will help you decide where your new content should live. For example, you would want to avoid putting a page written for experienced users between the already existing **Install** and **Basics** sections.

- What type of content am I creating?
- Who is my intended audience?
- Does this content require a new page, or can it be added to an existing page?
- If a new page or section must be added, how will it fit into the existing structure and flow of the docs?

When you've answered the questions outlined above and know where the new content will live, continue with the steps below.
```

> ### Fork, clone, and branch
> 
> You can't just make changes to the main repository directly, first you need to:
> 
> 1. Fork the repo.
> 2. Clone the repo to your local storage.
> 
> Follow [this guide](https://docs.ipfs.tech/community/contribute/contribution-tutorial/#creating-a-fix) to do the steps listed above.
> 
> Once you have the repo forked and cloned, the next step is to make a branch to commit all of your changes to. This will help keep your work organized, and will help you avoid committing undesired changes to your main branch. Here are the steps to do this in VSCode:
> 
> 1. Ensure that you are in the `ipfs-docs` directory on VSCode.
> 2. Click the branch icon in the bottom left corner of the VSCode window. A text field will open at the top of the window.
>  ![Create a new branch](./images/create-new-branch.png)
> 3. Type the desired name of your new branch. Preferably, the name of the branch reflects the work that you will be doing on the branch.
> 4. Once you have the name figured out, click **+Create new branch**.
> 
> Make sure that any changes you make are done in this new branch. You can verify which branch you are working on by checking the current branch in the bottom left corner of the window.

Reviewer: ```suggestion
### Fork, clone, and branch

You can't just make changes to the `main` branch directly. First, you need to:

1. Fork the repo.
2. Clone the repo to your local storage.

Follow [this guide](https://docs.ipfs.tech/community/contribute/contribution-tutorial/#creating-a-fix) to do the above steps.

Once you have created a fork, the next step is to make a branch to commit all of your changes to. Doing so will help keep your work organized and will help you avoid committing undesired changes to your main branch. Here are the steps to do this in VSCode:

1. Ensure that you are in the `ipfs-docs` directory on VSCode.
2. Click the branch icon in the bottom left corner of the VSCode window. A text field will open at the top of the window:

    ![Create a new branch](./images/create-new-branch.png)

3. Type the desired name of your new branch. Preferably, the name of the branch reflects the work that you will be doing on the branch.
4. Once you have the name figured out, click **Create new branch**.

Make sure that any changes you make are done in this new branch.
```

> ### Adding a single page
> 
> Adding a single page to an already existing section is as simple as adding one line to the `config.js` file.
> 
> When adding the new line, be sure to follow the exact same format and syntax as the other lines that are already present.
> 
> 1. Open `config.js` in VSCode.
> 2. Find the section you want to add a new page to. For this example, a page titled **Add a new page** will be added to the **Basics** section.
>  ![Add a new page](./images/add-new-page.png)
> 3. Add the new line to the desired section.
>   ```javascript
>   'title-of-section/title-of-new-page'
>   ```
> 4. With the page now added to `config.js`, you can move onto the [**commit your changes** section](#commit-your-changes).

Reviewer: ```suggestion
### Adding a single page

Adding a single page to an existing section is as simple as adding one line to the `config.js` file.

When adding the new line, follow the exact same format and syntax as the other lines already present.

1. Open `config.js`.
2. Find the section you want to add a new page to. For this example, a page titled **Add a new page** will be added to the **Basics** section:

    ![Add a new page](./images/add-new-page.png)

3. Add the new line to the desired section.

    ```javascript
    'title-of-section/title-of-new-page'
    ```

4. With the page now added to `config.js`, you can move on to the [**commit your changes** section](#commit-your-changes).
```

> ### Adding a new section
> 
> Adding a new section to the sidebar of the docs is a little more in-depth. For this example, a new section titled **Test** will be added between the **Basics** and **Concepts** sections.
> 
> When adding a new section, be sure you have the opening and closing curly brackets (`{` `}`) in the correct column. The easiest way to do this is to follow the same positioning and format as the other sections that are already in the file.
> 
> 1. Open `config.js` in VSCode.
> 2. Find the spot that you wish to add a new section. Again, for this example, the spot is between the **Basics** and **Concepts** sections.
>  ![Basics and concepts sections](./images/basics-and-concepts-sections.png)
> 3. Once you have found the right spot, add in the following lines, replacing any variables with your own content.
>  ![Add a new section](./images/add-new-section.png)

Reviewer: ```suggestion
### Adding a new section

Adding a new section to the sidebar of the docs is a little more in-depth. For this example, a new section titled **Test** will be added between the **Basics** and **Concepts** sections.

When adding a new section, be sure you have the opening and closing curly brackets (`{` `}`) in the correct column. The easiest way to do this is to follow the exact positioning and format of the other sections already in the file.

1. Open `config.js` in VSCode.
2. Find the spot where you wish to add a new section. Again, the spot is between the **Basics** and **Concepts** sections for this example:

    ![Basics and concepts sections](./images/basics-and-concepts-sections.png)

3. Once you have found the right spot, add the following lines, replacing any variables with your own content:

    ![Add a new section](./images/add-new-section.png)
```

> ### Commit your changes
> 
> Now that you have made the changes to `config.js`, you can commit them to your branch and make a pull request.
> 
> 1. Save the changes you have made to the `config.js` file.
> 2. Click the **source control** button in the VSCode sidebar.
>  ![Source control](./images/source-control.png)
> 3. In the **source control** menu, mouseover the changes you made and click the **+** button to stage those changes.
> 4. Now, click the **Commit** button to commit your changes to your branch.
> 5. Click the **Sync changes** button to push the changes back to GitHub.
> 
> The changes you made are now synced with your fork on GitHub. Now, you need to make a pull request in the `ipfs-docs` repo on GitHub. You can [follow these steps to create a pull request](https://docs.ipfs.tech/community/contribute/contribution-tutorial/#creating-a-fix).
> 
> The branch, pull request, and new pages or sections are now ready. The next step is to add the actual content of the new pages or sections to the pull request. Any changes you commit to the branch you just created will be added to that pull request.

Reviewer: ```suggestion
### Commit your changes

Now that you have made the changes to `config.js`, you can commit them to your branch and make a pull request.

1. Save the changes you have made to the `config.js` file.
2. Click **source control** in the VSCode sidebar:

    ![Source control](./images/source-control.png)

3. In the **source control** menu, mouseover the changes you made and click **+** to stage those changes.
4. Click **Commit** to commit your changes to your branch.
5. Click **Sync changes** to push the changes back to GitHub.

The changes you made are now synced with your fork on GitHub. You now need to make a pull request in the `ipfs-docs` repo on GitHub. You can [follow these steps to create a pull request](https://docs.ipfs.tech/community/contribute/contribution-tutorial/#creating-a-fix).

The branch, pull request, and new pages or sections are now ready. The next step is to add the actual content of the new pages or sections to the pull request. Any changes you commit to the branch you just created will be added to that pull request.
```

## Create a new page

Sometimes, you will need to make a whole new page for content to live. The process to do this is outlined below.

These steps were performed in Visual Studio Code (VSCode).

It's important to know what kind of content you are creating. The questions below will help you decide where your new content should live. For example, you would want to avoid putting a page written for experienced users between the already existing **Install** and **Basics** sections.

- What type of content am I creating?
- Who is my intended audience?
- Does this content require a new page, or can it be added to an existing page?
- If a new page or section must be added, how will it fit into the existing structure and flow of the docs?

When you've answered the questions outlined above and you know where the new content will live, continue with the steps below.

### Fork, clone, and branch

You can't just make changes to the main repository directly, first you need to:

> 1. Fork the repo.

Reviewer 2: Let's use "repository" to keep consistent with "repository" vs. "repo".

2. Clone the repo to your local storage.

Follow [this guide](https://docs.ipfs.tech/community/contribute/contribution-tutorial/#creating-a-fix) to do the above steps.

Once you have the repo forked and cloned, the next step is to make a branch to commit all of your changes to. Doing so will help keep your work organized and will help you avoid committing undesired changes to your main branch. Here are the steps to do this in VSCode:

1. Ensure that you are in the `ipfs-docs` directory on VSCode.
2. Click the branch icon in the bottom left corner of the VSCode window. A text field will open at the top of the window.

> 

Author: Not sure how I feel about this one. I could be wrong, but I don't think GitHub has support for tabs like this. This doc doesn't get published to the IPFS docs site, and only lives here on GitHub.

As for adding in instructions for doing this with the CLI, I don't think we need to get into that with this guide. The existing doc that I link to in the guide, https://docs.ipfs.tech/community/contribute/contribution-tutorial/#creating-a-fix, I believe already has the CLI git commands instructions that you're talking about.

@johnnymatthews do you have any thoughts on this?

> 

Reviewer: > explicitly mentioning the CLI git commands

GitHub doesn't support tabs in markdown, no. But @DannyS03 has a point; let's keep this guide software-agnostic. So, don't mention VSCode, just talk about the git commands required to get this working.

> 

Reviewer 2: What I meant with the tabs was using the tabs in vuepress to separate the content, as in:

| Text editor | CLI |

The same plugin used [here](https://docs.ipfs.tech/basics/js/js-ipfs/#install-js-ipfs).

> 

Reviewer 2: Regarding VSCode, the guide doesn't necessarily have to be rewritten, but abstracted with the same comment at the beginning about how the "text editor" being shown in the guide is VSCode, adding that there are others with similar capabilities.

> 

Author: New version available now! I removed the steps specific to VSCode and changed them to the CLI command. At the top I now mention that the screenshots I use to show the example changes in `config.js` are still screenshots of VSCode, but that these steps are the same, regardless of text editor used.

> You can also contribute to the project without specifically writing content for the site. Many discussions are happening all around the project. Protocol Labs, and by extension, IPFS, is a collaborative company that relies heavily on the community. We need your opinions and input! If you have an idea about a topic or want to weigh in on a discussion, feel free to have your say either on [GitHub](https://github.com/ipfs/ipfs-docs/issues) or on the [IPFS forums](https://discuss.ipfs.tech/).

Reviewer 3: Phrasing:
"Either" and "on" should be flipped. So "...to have your say on either GitHub or..."

## Create a new page

> Sometimes you will need to make a whole new page for content to live. The process to do this is outlined below.

Reviewer 3: Comma needed between "Sometimes" and "you" i.e. "Sometimes, you will need to..."

These steps were performed in Visual Studio Code (VSCode).

> It is important to know what kind of content you are creating. The questions below will help to determine where your new content should live. For example, you wouldn't want to put a page written for experienced users between the already existing **Install** and **Basics** sections.

Reviewer 3: Vale linter suggestion:
Use "It's" instead of "It is"

- What type of content am I creating?
- Who is my intended audience?
- Does this content require a new page or can it be added to an existing page?
- If a new page or section must be added, how will it fit into the existing structure and flow of the docs?

> When you have the above questions figured out and you know where the new content will live, continue on with the steps below.

Reviewer 3: Phrasing: "When you have the above questions..." is probably not the easiest way to word this.

Suggestion: Something like "When you've answered the questions above..."

### Fork, clone, and branch

> You can't just make changes to the main repo directly. First you need to fork the repo, from there you can clone it to your local machine to make changes to the files locally. Follow [these steps from the existing guide](https://docs.ipfs.tech/community/contribute/contribution-tutorial/#creating-a-fix) to do so. Once you have the repo forked and cloned, return here to continue on with creating a new page or section.

Reviewer 3: Suggestion: Break out the steps into numeric steps for readability. Something like:
```
1. [Fork the repo](https://docs.ipfs.tech/community/contribute/contribution-tutorial/#creating-a-fix)
2. Clone the repo using the steps from described in the existing guide (see the above link)
```

> You can't just make changes to the main repo directly. First you need to fork the repo, from there you can clone it to your local machine to make changes to the files locally. Follow [these steps from the existing guide](https://docs.ipfs.tech/community/contribute/contribution-tutorial/#creating-a-fix) to do so. Once you have the repo forked and cloned, return here to continue on with creating a new page or section.

Reviewer 3: Line 137 
Vale linter:
Use "continue" instead of "continue on" i.e. "...continue with the..." (Microsoft.Wordiness)

Before you start making changes to the files, make a branch to commit all of your changes to. This will help keep your work organized, and will help you avoid committing undesired changes to your main branch.
To do this in VSCode, follow the steps below.

> 1. Ensure you are in the `ipfs-docs` directory on VSCode.

Reviewer 3: "Ensure that you are in..." instead of "Ensure you are in..."

2. Click the branch icon in the bottom left corner of the VSCode window.
 ![Create a new branch](./images/create-new-branch.png)

> 1. A text field will open at the top of the window, type the desired name of your new branch. Preferably, the name of the branch reflects the work that you will be doing on the branch.

Reviewer 3: The "A text field will open at the top of the window" comment should probably be the second sentence in the "Click the branch icon..." step above since it's a result of that action.

Then, the next steps begins "Type the desired name of..."

1. Once you have the name figured out, click **+Create new branch**.

> Make sure that any changes you make are done in this new branch, you can verify this by checking the current branch in the bottom left corner of the window.

Reviewer 3: Split this into two sentences for readability.

`Make sure that any changes you make are done in the new branch. You the branch that you are by checking the bottom left corner of the window.` or something like that

### Adding a single page

Adding a single page to an already existing section is as simple as adding one line to the `config.js` file.

When adding the new line, be sure to follow the exact same format and syntax as the other lines that are already present.

1. Open `config.js` in VSCode.
1. Find the section you want to add a new page to. For this example, a page titled **Add a new page** will be added to the **Basics** section.
 ![Add a new page](./images/add-new-page.png)
1. Add the new line to the desired section.

>   ```

Reviewer 3: Code blocks should indicate the language for consistency. So, probably javascript for this code block since it's in `config.js`

  'title-of-section/title-of-new-page'
  ```

> 4. With the page now added to `config.js`, save the file, stage the changes, and commit them to your branch.

Reviewer 3: Would suggest breaking these out into individual steps

### Adding a new section

Adding a new section to the sidebar of the docs is a little more in-depth. For this example, a new section titled **Test** will be added between the **Basics** and **Concepts** sections.

When adding a new section, be sure you have the opening and closing curly brackets (`{` `}`) in the correct column. The easiest way to do this is to follow the same positioning and format as the other sections that are already in the file.

> 1. Open `config.js` in VSCode.

Reviewer 3: List numbering: for consistency / readability if someone is wanting to actually read straight Markdown, we should pick a numbered list format, either:
```
1.
1.
1.
```
or
```
1.
2.
3.
```
However, I do not think we should mix the two formats. This comment applies to all ordered lists in this PR.

Author: I seem to remember @johnnymatthews telling me numbered lists should be created using the
```
1.
1.
1.
```
format.

As for the issues present in this PR, I believe they are caused by VSCode not liking the "all 1's" format. When an element is inserted between two of the list items, like an image link or code blocks, VSCode automatically changes the following number to a 2 or 3 or whatever it thinks should come next.

I'll go through and make sure they are all consistent.

1. Find the spot that you wish to add a new section. Again, for this example, the spot is between the **Basics** and **Concepts** sections.
 ![Basics and concepts sections](./images/basics-and-concepts-sections.png)
3. Once you have found the right spot, add in the following lines, replacing any variables with your own content.
 ![Add a new section](./images/add-new-section.png)

### Commit your changes

Now that you have made the changes to `config.js`, you can commit them to your branch and make a pull request.

1. Save the changes you have made to the `config.js` file.
1. Click the **source control** button in the VSCode sidebar.
 ![Source control](./images/source-control.png)

> 2. In the **source control** menu, mouse-over the changes you made and click the **+** button to stage those changes.

Reviewer 3: I think "mouse-over" should not have a hyphen - haven't confirmed this though

Author: As far as I can tell, mouseover is, in fact, one word.

> 3. Now click the **Commit** button to commit your changes to your branch.

Reviewer 3: Comma between "Now" and "Click"

`Now, click the...`

4. Click the **Sync changes** button to push the changes back to GitHub.

> The changes you made are now synced with your fork on GitHub. Now you need to make a pull request in the `ipfs-docs` repo on GitHub. You can [follow these steps to create a pull request](https://docs.ipfs.tech/community/contribute/contribution-tutorial/#creating-a-fix).

Reviewer 3: Comma between "Now" and "you" i.e. "Now, you need to make..."

> The branch, pull request, and new pages or sections are now ready. You can continue on with adding the actual content of the new pages or sections to the pull request. Any changes you commit to the branch you just created will be added to that pull request.

Reviewer 3: Vale linter:
Use "continue" instead of "continue on" i.e. "You can continue adding the actual..." (Microsoft.Wordiness)

You can also contribute to the project without specifically writing content for the site. Many discussions are happening all around the project. Protocol Labs, and by extension, IPFS, is a collaborative company that relies heavily on the community. We need your opinions and input! If you have an idea about a topic or want to weigh in on a discussion, feel free to have your say either on [GitHub](https://github.com/ipfs/ipfs-docs/issues) or on the [IPFS forums](https://discuss.ipfs.tech/).

## Create a new page

Sometimes you will need to make a whole new page for content to live. The process to do this is outlined below.

These steps were performed in Visual Studio Code (VSCode).

> The first step to creating a new page is to identify the type of content it will be to determine where it will live in the structure of the docs. It may need to be a whole new section or it could be added to an existing section as a sub-page. An important concept to keep in mind when deciding where to put the new content is the flow of the already existing content. You wouldn't want to put a page or section designed for experienced users between the already existing **Install** and **Basics** sections.

Reviewer 3: I like that you're going into all the things that the contributor should think about before adding a page. You could rephrase it as a series of questions. Something like:
```
Before creating a new page, ask yourself the following questions:

- What type of content am I creating? A tutorial, a conceptual explanation, or a few grammatical edits?
- Who is my intended audience? New developers, seasoned Web3 veterans, or the general community?
- Does this content require a whole new page or can it be condensed into a section or sub-page?
- Can an existing page have the new content added to it, instead of creating a whole new page?
- If a new page must be added, how will it fit into the overall flow and structure of the existing documentation?

Thinking about these questions before creating a new page benefits you and the PLN community. For example, putting a page or section designed for experienced users between the current **Install** and **Basics** sections might negatively impact the documentation experience for the community.
```

Author: Agree, I think this works better. Implemented in the newest version.

> Once you have determined where the content will live, you must add the appropriate lines to the `config.js` file.

Reviewer 3: The flow here is confusing me somewhat. LMK if I'm misunderstanding here. 

_Thoughts:_
Line 128 says ```Once you have determined where the content will live, you must add the appropriate lines to the `config.js` file.```. Then, the reader is directed to create a new branch, which makes sense. Then, the reader adds the line for the page to config.js. 

_Suggestion:_
Seems like it would make more sense for line 128 to lead into the process of creating a new branch and the current line 128 content can be moved to the bottom of the "Create a branch" section to lead into the "Adding a single page" section.

Reviewer 3: (Sorry, I realized that this comment highlights the wrong lines)

Author: Addressed this in the new version as well.

> ### Create a branch

Reviewer 3: This is more of a question / discussion: should we be directing the user to just branch off the main repo and work off of that? Or should we request that they create a fork so that they can make edits there, create a PR, and then we review, merge, etc.? Just wondering if there is preference or opinions on over the other. I like forking because it means way less branches on the main repo, so my fork can be messy and it doesn't affect the overall repo quality. Would like to hear other folks take on this. @TMoMoreau I see that we already have https://docs.ipfs.tech/community/contribute/contribution-tutorial/#creating-a-fix and https://docs.ipfs.tech/community/contribute/contribution-tutorial/#create-a-pull-request. Might make sense to just tell the user to follow the create fix process, with the added note that their edits must include the necessary edits to `config.js` file (depending on whether they are adding a section or page). Then, have them follow the PR process.

Reviewer 3: Seems like this might cause confusion, especially if a reader reads both this and the existing guidelines in the published docs.

Reviewer 3: Unless I'm totally misunderstanding what I'm reading here (totally could be).

Reviewer 3: Steps could be:

```
1. Complete steps 1-3 of https://docs.ipfs.tech/community/contribute/contribution-tutorial/#creating-a-fix if not already completed. Could either link the user or copy out that MD to this page.
2. Make local change. Could be broken into substeps or their individual steps: 
     1. Create page, section, grammar fix, whatever. 
     2. Make appropriate edits to config.js depending on whether it's a new section or page. 
         Could probably condense https://github.com/ipfs/ipfs-docs/pull/1294/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055R146-R173 into one section. 
3. https://docs.ipfs.tech/community/contribute/contribution-tutorial/#create-a-pull-request
4. https://docs.ipfs.tech/community/contribute/contribution-tutorial/#waiting-for-a-review
5. etc.
```

Reviewer: It'd be nice if contributors just made forks. Like you said @ElPaisano, individual forks can be as messy as they want, until it comes time to make a PR.

Author: Reader is now instructed to first fork the repo, rather than cloning the main repo. They are also directed to our existing docs on how to create a fix/PR.

You can't just add a new page directly to the main branch. First, you will have to create your own branch to push commits to.

To do this in VSCode:

1. Ensure you are in the `ipfs-docs` directory on VSCode.
1. Click the branch icon in the bottom left of the window.
    <br><br>
    <img src="./images/create-new-branch.png" style="display:block;margin:0 auto;width:40vw;">

>     <br>

Reviewer: Why not us the markdown image format? If you don't like they way markdown images are showing up, you should change how images are deal with in the CSS.

Adding HTML into markdown gets messy really quick.

Author: I'll change it to the markdown format. I used HTML to have control over size, but the default size might be okay.

Reviewer: You could also just resize the images manually and then re-upload them if you wanted. Depends on what you're trying to resize the image to I guess.

Author: Resized the images manually and changed them to markdown image format. Some of them may still be too big in size, I can adjust later if that is the case.
